extensions/oni-plugin-quickopen/src/QuickOpen.ts

```diff
@@ -0,0 +1,435 @@
+                const timeout = window.setTimeout(() => {
+                    this._menu.setLoading(true)
+                }, 200)
+                this._searcher.changeQueryText(newFilter)
+                if (timeout) {
+                    window.clearTimeout(timeout)
+                }
+            } else {
+                this._searcher.cancel()
+            }
+        })
+
+        this._menu.onItemSelected.subscribe((selectedItem: any) => {
+            this.open(this.getDefaultOpenMode())
+        })
+    }
+
+    public isOpen(): boolean {
+        return this._menu.isOpen()
+    }
+
+    public async open(mode: Oni.FileOpenMode): Promise<void> {
+        const selectedItem = this._menu.selectedItem
+        if (!selectedItem) {
+            return
+        }
+
+        if (isType(selectedItem, QuickOpenType.bookmarkHelp)) {
+            this._oni.commands.executeCommand("oni.config.openConfigJs")
+        } else if (isType(selectedItem, QuickOpenType.folderHelp)) {
+            this._oni.commands.executeCommand("workspace.openFolder")
+        } else if (isType(selectedItem, QuickOpenType.bufferLine)) {
+            if (mode !== Oni.FileOpenMode.Edit) {
+                await this._oni.editors.openFile(
+                    this._oni.editors.activeEditor.activeBuffer.filePath,
+                    {
+                        openMode: mode,
+                    },
+                )
+            }
+            await this._oni.editors.activeEditor.neovim.command(`${selectedItem.label}`)
+        } else {
+            this._seenItems.push(selectedItem.metadata["hash"])
+
+            const { activeWorkspace } = this._oni.workspace
+            const pathArgs = (activeWorkspace ? [activeWorkspace] : []).concat([
+                selectedItem.metadata["path"],
+            ])
+            const fullPath = path.join(...pathArgs).replace("~", getHome())
+            await this._oni.editors.openFile(fullPath, { openMode: mode })
+
+            if (isType(selectedItem, QuickOpenType.folder)) {
+                this._oni.workspace.changeDirectory(fullPath)
+            } else if (isType(selectedItem, QuickOpenType.bookmark)) {
+                // If we are bookmark, and we open a file, the open it's dirname
+                // If we are a directory, open it.
+                const dirPath = lstatSync(fullPath).isDirectory() ? fullPath : selectedItem.detail
+                this._oni.workspace.changeDirectory(dirPath)
+            }
+        }
+
+        this._menu.hide()
+    }
+
+    public async setToQuickFix() {
+        this._oniNext.populateQuickFix(this._itemsFound.map(item => item.toQuickFixItem()))
+        this._menu.hide()
+    }
+
+    public openFileWithAltAction(): void {
+        const mode: Oni.FileOpenMode = this._oni.configuration.getValue(
+            "editor.quickOpen.alternativeOpenMode",
+        )
+        this.open(mode)
+    }
+
+    public async searchFileByContent() {
+        const filterName = "none"
```

Comment:
Should this be a config option?

My first instinct was to try and use smart search (ie case insenstive unless you use uppercase characters), which then failed.

Reply:
This `"none"` is intentional.
This `filter` field is a second filtering done after `rg`.
Since `rg` can accept a regex, a search string like `ab.*cd` will be computed as `rg ab.*cd | egrep "ab\.\*cd"` which will very likely cause similar searches the yield no results.

However:
1. Found the reason that smart-case didn't work and fixed it.
2. Added a TODO to use a filter for highlighting w/ support for regex.

I tried the `regexFilter` for the menu, but it throws some exceptions.
It might be that a slight update/variation can fix it. (But I prefer to do less changes that will keep this PR further for merging)

(Will push soon)

Reply:
P.S: fixed the latency/hanging issue.